contracts/SpokePoolV3Periphery.sol

```diff
@@ -551,6 +644,18 @@ contract SpokePoolV3Periphery is Lockable, MultiCaller {
         _depositV3(acrossInputToken, acrossInputAmount, depositData);
     }
 
+    /**
+     * @notice Verifies that the signer is the owner of the signing contract.
+     * @dev The _hash and _signature fields are intentionally ignored since this contract will accept
+     * any signature which originated from permit2 after the call to the exchange.
+     * @dev This is safe since this contract should never hold funds nor approvals, other than when it is depositing or swapping.
+     */
+    function isValidSignature(bytes32, bytes calldata) external view returns (bytes4 magicBytes) {
```

Comment:
Can we actually just always return true? My reasoning is that this contract can already perform arbitrary calls -- it should _never_ have tokens or approvals -- and presumably the only thing this is used for is to transfer funds from this contract to an exchange while in the middle of a swap transaction.

Reply:
i think you're correct in checking `expectingPermit2Callback`, but want to see what others think